### ISSUE TITLE
Improve usage of yargs

### ISSUE THREAD

**Issue by jormaechea** [design2soft/memorize#4]
Hey, I recommended you to use `yargs` in reddit a while ago..
So, here are some comments on how you can improve it:

### 1. help
I saw you added your custom help message, which can turn into a pain to mantain. Yargs has some built-in help, which you used, but didn't took 100% of it's greatness.

You can remove all this custom code, and the `showHelp()` helper:
https://github.com/design2soft/memorize/blob/master/index.js#L7-L12

And you can add the following to the yargs call chain:
```js
  .demandCommand(1, 'You need to pass a command!')
```

That way, it will be validated and commands will be included in the help message. [Docs](https://github.com/yargs/yargs/blob/HEAD/docs/api.md#demandcommandmin1-minmsg)

### 2. commands loading
You did something good that's putting all the comands in a `commands/` dir. Now you can level your game up and change every `.command()` calls by a single `.commandDir()` call! [Docs](https://github.com/yargs/yargs/blob/HEAD/docs/api.md#commanddirdirectory-opts)

### 3. Command arguments
I both commands, you made a manual validation of their arguments. You can give `yargs` the responsibility to do that by configuring you commands a little bit more. You can _demand_ an option. [Docs](https://github.com/yargs/yargs/blob/HEAD/docs/api.md#demandoptionkey-msg)

### 4. Bonus track
I found something strange in the commands definition:
```js
(exports.command = "get [key]"), "Get a saved information value";
```

Couldn't that just be `exports.command = "get [key]";`?

That's like that both in get and set commands.

---

Happy coding!

**Comment by madhavmalhotra-3089** [comment]
Hey @jormaechea Have made some of the suggested changes, but I could not quite figure out how to use the `demandOption`. Do you think you will have time to take a look though and also help me through that code?

**Comment by jormaechea** [comment]
Yeah, sure! I didn't find this well documented, but I've done it some time ago.

`yargs` allows you to export a _builder_ for each command, where you can configure it. For example, your `get` command can be something like this:

```js
const helpers = require("../helper");
const { memoryFile } = require("../constants");

exports.command = "get [key]";
exports.describe = "Get a saved information value";
exports.builder = yargs => yargs.demandOption('key');
exports.handler = ({ key }) => helpers.get(memoryFile, key);
```

Hope it helps!
